packages/core/src/define-hook.ts

```diff
@@ -32,7 +34,7 @@ import { resumeHook } from './runtime/resume-hook.js';
  * }
  * ```
  */
-export function defineHook<T>() {
+export function defineHook<T>(schema?: ZodType<T>) {
```

Comment:
I feel like the schema should specified as a `schema` prop in an options object, to avoid a breaking change in case we decide to add more props in the future.